src/engine/segment.go

```diff
@@ -399,6 +402,9 @@ func (segment *Segment) SetEnabled(env platform.Environment) {
 			}
 		}
 	}
+	if width, _, err := term.GetSize(int(os.Stdin.Fd())); err == nil && segment.MaxCols > 0 && width > segment.MaxCols {
```

Comment:
```suggestion
	if width, err := segment.env.TerminalWidth(); err == nil && segment.MaxCols > 0 && width > segment.MaxCols {
```
We actually have this number already for other means so no use for a specific library here. Make sure to use `go mod tidy` after applying this change to remove the dependencies.